res/controllers/Pioneer-DDJ-SX-scripts.js

```diff
@@ -1571,6 +1572,8 @@ PioneerDDJSX.panelSelectButton = function(channel, control, value, status, group
 
 PioneerDDJSX.shiftPanelSelectButton = function(channel, control, value, status, group) {
     var channelGroup;
+    PioneerDDJSX.shiftPanelSelectPressed = value ? true : false;
```

Comment:
If I understand correctly, you need to hold shift, hold panel, and press an effect assignment button to toggle assignment for effect units 3 & 4? Don't you need 3 hands for that? This is why I suggested moving the old Panel functionality to shift + Panel, so you could hold the Panel button with one hand and press the effect unit assignment button with the other hand.